script.js

```diff
@@ -1,12 +1,434 @@
+    return show.name.toLowerCase().includes(searchString);
+  });
+  const allShowsCard = createCardForEachShow(filteredEpisodes);
+  allShowPopUp.appendChild(allShowsCard);
+});
+
+const onclickResetPage = () => {
+  allShowPopUp.innerHTML = "";
+};
+
+// <---search bar anime function--->
+const expand = () => {
+  searchBtn.classList.toggle("close");
+  input.classList.toggle("square");
+  currentEpisodes.innerText = "";
+  displayEpisodes(showEpisodes);
+};
+
+searchBtn.addEventListener("click", expand);
+
+//<----------------create random card ----------------->
+
+const createRandomCard = (show) => {
+  let htmlElement = ` <div> <img src="${
+    show.image !== null ? show.image.original : ""
+  }" width="80%" height="1000px" alt="Cover poster of ${
+    show.name
+  }" id="randomCard-img">
+      <div id="randomCard-info">
+      <h1 id="randomCard-title">${show.name}</h1>
+      <p><span id="span">Language: ${show.language}</span></br>
+      ${show.genres}</br>Rating: ${show.rating.average}</p>
+      <div id="btn-flex">
+      <button id="watch-now"><a href="${
+        show.url
+      }" target="_blank"><div class="play"></div>Watch Now</a></button>
+         <div class="box">
+	<a class="info-btn-popup" href="#popup1"><i class="fa fa-info-circle" style="font-size:24px"></i>More info</a>
+</div>
+     </div> </div></div>`;
+
+  let popupInfo = `${show.summary !== null ? show.summary : ""}`;
+  popUpInfo.innerHTML = popupInfo;
+  randomCard.innerHTML = htmlElement;
+};
+
+const randomFile = (showFile) => {
+  let chooseRandomShow = showFile[Math.floor(Math.random() * showFile.length)];
+  createRandomCard(chooseRandomShow);
+};
+
+//<---------Code below this is Create Category for all show (carousel slider)-------------------->
+//<------------------function to catagories shows according to the type-------------->
+const findTypeOfShow = (type) => {
+  return newShows.filter((show) => show.type.includes(type));
+};
+
+//<------function to catagories shows according to the genres------------>
+const findGenres = (genres) => {
+  return newShows.filter((show) => show.genres.includes(genres));
+};
+
+// <------------------------Top rated series by rating-------------------------------->
+const topRated = newShows.filter((category) => {
+  const network = `${category.rating.average}`;
+  if (network >= 9) {
+    return category;
+  }
+});
+const topRatedContainer = document.getElementById("top-rated");
+const topCard = createCardForEachShow(topRated);
+topRatedContainer.appendChild(topCard);
+//<-----------------------------British films and Tv series------------------------------->
+const findGB = newShows.filter((category) => {
+  const network = `${
+    category.network !== null ? category.network.country.code : ""
+  }`;
+  if (network === "GB") {
+    return category;
+  }
+});
+
+const gbContainer = document.getElementById("british");
+const gbCard = createCardForEachShow(findGB);
+gbContainer.appendChild(gbCard);
+
```

Comment:
brilliant skills and code style and appreciate that you have excelled and demonstrated knowledge and extended to advanced levels